test/node-http-test.js

```diff
@@ -0,0 +1,283 @@
+    it('It should return a proof type of TYPE_DEADEND when an auction has not been initiated', async () => {
+      const proof = await nclient.get(`/proof/name/${NAME0}`);
+      assert.equal(proof.proof.type, 'TYPE_DEADEND');
+      assert.equal(proof.name, NAME0);
+    });
+
+    describe('When an auction has been initiated', () => {
+      it('It should return the name\'s proof', async () => {
+        await mineBlocks(10, cbAddress);
+
+        await wclient.execute('sendopen', [NAME0]);
+        await mineBlocks(1, cbAddress);
+
+        const { stats: { blocksUntilBidding } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilBidding, cbAddress);
+
+        const sendBid = await wclient.execute('sendbid', [NAME0, 12, 12]);
+        assert(sendBid);
+        const { stats: { blocksUntilReveal } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilReveal, cbAddress);
+
+        await wclient.execute('sendreveal', [NAME0]);
+        const { stats: { blocksUntilClose } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilClose, cbAddress);
+
+        await wclient.execute('sendupdate', [NAME0, { compat: false, version: 0, ttl: 172800, ns: ['ns1.example.com.@1.2.3.4'] }]);
+        await mineBlocks(1, cbAddress);
+
+        const proof = await nclient.get(`/proof/name/${NAME0}`);
+        assert.equal(proof.proof.type, 'TYPE_EXISTS');
+        assert.equal(proof.name, NAME0);
+      });
+    });
+  });
+
+  describe('getNameProofByHash', () => {
+    it('It should return a proof type of TYPE_DEADEND when an auction has not been initiated', async () => {
+      const nameHash = rules.hashName(NAME0).toString('hex');
+      const proof = await nclient.get(`/proof/hash/${nameHash}`);
+      assert.equal(proof.proof.type, 'TYPE_DEADEND');
+      assert.equal(proof.name, null);
+    });
+
+    describe('When an auction has been initiated', () => {
+      it('It should return the name\'s proof', async () => {
+        await mineBlocks(10, cbAddress);
+
+        await wclient.execute('sendopen', [NAME0]);
+        await mineBlocks(1, cbAddress);
+
+        const { stats: { blocksUntilBidding } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilBidding, cbAddress);
+
+        const sendBid = await wclient.execute('sendbid', [NAME0, 12, 12]);
+        assert(sendBid);
+        const { stats: { blocksUntilReveal } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilReveal, cbAddress);
+
+        await wclient.execute('sendreveal', [NAME0]);
+        const { stats: { blocksUntilClose } } = await wclient.execute('getauctioninfo', [NAME0]);
+
+        await mineBlocks(blocksUntilClose, cbAddress);
+
+        await wclient.execute('sendupdate', [NAME0, { compat: false, version: 0, ttl: 172800, ns: ['ns1.example.com.@1.2.3.4'] }]);
```

Comment:
This line is a little too long, can you turn the record into its own object? It would also be nice to assert that getting the namestate reflects the record.